app/src/androidTest/java/com/github/sdp/mediato/SearchFragmentTest.java

```diff
@@ -175,6 +177,61 @@ public void testFollowAndUnfollow() {
     assertDisplayedAtPosition(R.id.searchactivity_recyclerView, 0, R.id.userAdapter_userName, user2.getUsername());
     assertDisplayedAtPosition(R.id.searchactivity_recyclerView, 0, R.id.userAdapter_followButton, R.string.searchUser_follow);
   }
+
+  @Test
+  public void testMovieSearchWithEmptyString() {
+    clickOn(R.id.search_category_movie);
+    clickOn(androidx.appcompat.R.id.search_button);
+    typeTo(androidx.appcompat.R.id.search_src_text, "Potter");
+
+    sleep(700);
+
+    clearText(androidx.appcompat.R.id.search_src_text);
+    pressImeActionButton();
+
+    sleep(700);
+
+    assertRecyclerViewItemCount(R.id.searchactivity_recyclerView, 0);
+  }
+
+  @Test
+  public void testMovieSearchWithUnknownMovie() {
+    clickOn(R.id.search_category_movie);
+    clickOn(androidx.appcompat.R.id.search_button);
+    typeTo(androidx.appcompat.R.id.search_src_text, "jadvbipehsjdb");
+    pressImeActionButton();
+
+    sleep(1000);
+
+    assertRecyclerViewItemCount(R.id.searchactivity_recyclerView, 0);
+  }
+
+  @Test
+  public void testMovieSearchWithKnownMovie() {
+    clickOn(R.id.search_category_movie);
+    clickOn(androidx.appcompat.R.id.search_button);
+    typeTo(androidx.appcompat.R.id.search_src_text, "Harry Potter and the half blood prince");
+
+    sleep(500);
+    assertDisplayedAtPosition(R.id.searchactivity_recyclerView, 0, R.id.text_title, "Harry Potter and the Half-Blood Prince");
+  }
+
+  @Test
+  public void testClickOnMovieSearchResultOpensRatingScreen() {
+    clickOn(R.id.search_category_movie);
+    clickOn(androidx.appcompat.R.id.search_button);
+    typeTo(androidx.appcompat.R.id.search_src_text, "Harry Potter and the half blood prince");

```

Comment:
Would this work ? It might be more readable and logic. This change might be needed in all tests if it is a non-breaking one.

```suggestion
    clickOn(R.id.search_button);
    typeTo(R.id.search_src_text, "Harry Potter and the half blood prince");

```